dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java

```diff
@@ -1712,74 +1719,44 @@ boolean reportErrors(Result result, Collection<String> restrictedAccessAttempts,
-            if (!!!configuredAlreadyReported(configured1, configured2, reportedConfigured)) {
-                if (isEeCompatible(conflict1)) {
+            if (!!!configuredAlreadyReported(conflictRecord1.configured, conflictRecord2.configured, reportedConfigured)) {
+                if (isEeCompatible(conflictRecord1.conflict)) {
                     final boolean ignoreVersion = true;
-                    if (getEeCompatiblePlatform(conflict1, ignoreVersion).equals(getEeCompatiblePlatform(conflict2, ignoreVersion))) {
+                    if (getEeCompatiblePlatform(conflictRecord1.conflict, ignoreVersion).equals(getEeCompatiblePlatform(conflictRecord2.conflict, ignoreVersion))) {
                         // Both conflicting features support "Java EE X" or "Jakarta EE X", exclusively
-                        Tr.error(tc, "UPDATE_CONFLICT_INCOMPATIBLE_EE_FEATURES_SAME_PLATFORM_ERROR", getPreferredEePlatform(eeConflict1),
-                                 getPreferredEePlatform(eeConflict2), getFeatureName(configured1), getFeatureName(configured2), getEeCompatiblePlatform(conflict1, ignoreVersion));
+                        Tr.error(tc, "UPDATE_CONFLICT_INCOMPATIBLE_EE_FEATURES_SAME_PLATFORM_ERROR", getPreferredEePlatform(conflictRecord1.compatibleConflict),
+                                 getPreferredEePlatform(conflictRecord2.compatibleConflict), getFeatureName(conflictRecord1.configured), getFeatureName(conflictRecord2.configured),
+                                 getEeCompatiblePlatform(conflictRecord1.conflict, ignoreVersion));
                     } else {
                         // One conflicting feature supports "Jakarta EE X", the other "Java EE X"
-                        Tr.error(tc, "UPDATE_CONFLICT_INCOMPATIBLE_EE_FEATURES_DIFFERENT_PLATFORM_ERROR", getPreferredEePlatform(eeConflict1),
-                                 getPreferredEePlatform(eeConflict2), getFeatureName(eeConflict1), getFeatureName(eeConflict2), getFeatureName(configured1),
-                                 getFeatureName(configured2));
+                        Tr.error(tc, "UPDATE_CONFLICT_INCOMPATIBLE_EE_FEATURES_DIFFERENT_PLATFORM_ERROR", getPreferredEePlatform(conflictRecord1.compatibleConflict),
+                                 getPreferredEePlatform(conflictRecord2.compatibleConflict), getFeatureName(conflictRecord1.compatibleConflict),
+                                 getFeatureName(conflictRecord2.compatibleConflict), getFeatureName(conflictRecord1.configured),
+                                 getFeatureName(conflictRecord2.configured));
                         // Remove the conflicting features (not necessarily the configured features)
-                        result.getResolvedFeatures().remove(getFeatureName(eeConflict1));
-                        result.getResolvedFeatures().remove(getFeatureName(eeConflict2));
+                        result.getResolvedFeatures().remove(getFeatureName(conflictRecord1.compatibleConflict));
```

Comment:
_Unrelated to this review_: Removing features from the resolved set is unnecessary when disableAllOnConflict is true.

Reply:
I thought about that when adding `disableAllOnConflict` but thought keeping the code as-is there was more simple than determining if the remove is necessary or not.